### PR TITLE
New: Format table column based on user config

### PR DESCRIPTION
**Pull request by Xuefeng-Zhu**

### CODE REVIEW

**Comment by Xuefeng-Zhu** [comment]
@balajimanoharan I think it will be nice to have a hbs helper, but what is helper supposed to do and where should we use it?

**Comment by balajimanoharan** [comment]
I would suggest changing [this](https://github.com/yahoo/navi/blob/master/packages/visualizations/addon/templates/components/cell-renderers/metric.hbs#L2) file to 
```hbs
{{format-value metricValue format}}
```

and you could use the same thing for the threshold cell renderer

**Comment by Xuefeng-Zhu** [comment]
@balajimanoharan I have refactored the code with the helper

File: packages/visualizations/addon/helpers/format-table-cell.js [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+ * Copyright 2018, Yahoo Holdings Inc.
+ * Licensed under the terms of the MIT license. See accompanying LICENSE.md file for terms.
+ */
+
+import { helper } from '@ember/component/helper';
+import { isEmpty } from '@ember/utils';
+import numeral from 'numeral';
+import { smartFormatNumber } from 'navi-core/helpers/smart-format-number';
+
+export function formatTableCell([value, format, smartFormat]) {
```
Comment (balajimanoharan) [inline packages/visualizations/addon/helpers/format-table-cell.js]: Can we call the boolean argument `shouldSmartFormat`?
And can we add jsdoc for this function?

File: packages/visualizations/tests/integration/helpers/format-table-cell-test.js [hunk 2]
```diff
@@ -0,0 +1,34 @@
+
```
Comment (balajimanoharan) [inline packages/visualizations/tests/integration/helpers/format-table-cell-test.js]: Can we clean this up? extra `\n`

File: packages/visualizations/addon/helpers/format-table-cell.js [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+export function formatTableCell([value, format, shouldSmartFormat]) {
+  if (isEmpty(value)) {
+    return '--';
+  }
+
+  if (format) {
+    return numeral(value).format(format);
+  }
+
+  return shouldSmartFormat ? smartFormatNumber([value]) : value;
```
[the remark this answers is not in the document]
Comment (Xuefeng-Zhu) [reply packages/visualizations/addon/helpers/format-table-cell.js]: for threshold cell, we do not want smart format number

**Comment by Xuefeng-Zhu** [comment]
@jkusa I have updated the code based on discussion

File: packages/visualizations/addon/templates/components/cell-renderers/threshold.hbs [hunk 4]
```diff
@@ -1,2 +1,2 @@
-{{!-- Copyright 2017, Yahoo Holdings Inc. Licensed under the terms of the MIT license. See accompanying LICENSE.md file for terms. --}}
-{{metricValue}}
+{{!-- Copyright 2018, Yahoo Holdings Inc. Licensed under the terms of the MIT license. See accompanying LICENSE.md file for terms. --}}
+{{format-table-cell metricValue format}}
```
Comment (balajimanoharan) [inline packages/visualizations/addon/templates/components/cell-renderers/threshold.hbs]: Now, would it be better to extend the metric cell renderer for this? 
@jkusa what do you think?

Comment (jkusa) [reply packages/visualizations/addon/templates/components/cell-renderers/threshold.hbs]: @balajimanoharan maybe, can `threshold` extend `metric`?

Comment (balajimanoharan) [reply packages/visualizations/addon/templates/components/cell-renderers/threshold.hbs]: it probably could. there only one extra property in threshold, `valueIndicator`. Everything else is the same.

Comment (Xuefeng-Zhu) [reply packages/visualizations/addon/templates/components/cell-renderers/threshold.hbs]: @jkusa updated

Comment (jkusa) [reply packages/visualizations/addon/templates/components/cell-renderers/threshold.hbs]: Do we need the helper now?

Comment (Xuefeng-Zhu) [reply packages/visualizations/addon/templates/components/cell-renderers/threshold.hbs]: @jkusa Do you refer to the template file?

File: packages/visualizations/addon/templates/components/cell-renderers/metric.hbs [hunk 5]
```diff
@@ -1,2 +1,2 @@
-{{!-- Copyright 2017, Yahoo Holdings Inc. Licensed under the terms of the MIT license. See accompanying LICENSE.md file for terms. --}}
-{{metricValue}}
+{{!-- Copyright 2018, Yahoo Holdings Inc. Licensed under the terms of the MIT license. See accompanying LICENSE.md file for terms. --}}
+{{format-table-cell metricValue format}}
```
Comment (jkusa) [inline packages/visualizations/addon/templates/components/cell-renderers/metric.hbs]: Do we need this helper for a single use? Can we bring this into the metric cell-renderer

Comment (Xuefeng-Zhu) [reply packages/visualizations/addon/templates/components/cell-renderers/metric.hbs]: yeah. It seems that it is not needed anymore

Comment (Xuefeng-Zhu) [reply packages/visualizations/addon/templates/components/cell-renderers/metric.hbs]: @jkusa removed

**Pull request merged**